integration-test/pom.xml

```diff
@@ -135,7 +135,7 @@
               <executable>/bin/sh</executable>
               <arguments>
                 <argument>-c</argument>
-                <argument>rm -rf spark-distro; mkdir spark-distro-tmp; cd spark-distro-tmp; tar xfz ${spark-distro-tgz}; mv * ../spark-distro; cd ..; rm -rf spark-distro-tmp</argument>
+                <argument>rm -rf spark-distro; mkdir spark-distro; mkdir spark-distro-tmp; cd spark-distro-tmp; tar xfz ${spark-distro-tgz}; mv * ../spark-distro; cd ..; rm -rf spark-distro-tmp</argument>
```

Comment:
Hmm. I see this adding `mkdir spark-distro`. But later we go inside a tmp dir, untar the distro and do `mv * ../spark-distro`.

The unpacked tarball has a top level dir, like spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9. Wouldn't the `mv` command create a subdir hierarchy we don't want, like `spark-distro/spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9`?

Reply:
In my usage, I was actually seeing a failure without this create step. `mv * ../spark-distro` actually expects the directory to be created and present was my understanding.

Reply:
The intent was the `mv` statement will rename the unpacked top level dir to be the `../spark-distro` dir. I guess your tarball is not like my tarball :-) Can you check if your tarball creates a top-level dir?

Here's mine:
```
$ tar tvf spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9.tgz | head
drwxr-xr-x kimoonkim/staff   0 2017-12-18 13:09 spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9/
drwxr-xr-x kimoonkim/staff   0 2017-12-18 13:09 spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9/bin/
-rwxr-xr-x kimoonkim/staff 1089 2017-12-18 13:09 spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9/bin/beeline
-rw-r--r-- kimoonkim/staff 1064 2017-12-18 13:09 spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9/bin/beeline.cmd
-rwxr-xr-x kimoonkim/staff 1933 2017-12-18 13:09 spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9/bin/find-spark-home
-rw-r--r-- kimoonkim/staff 2681 2017-12-18 13:09 spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9/bin/find-spark-home.cmd
-rw-r--r-- kimoonkim/staff 1892 2017-12-18 13:09 spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9/bin/load-spark-env.cmd
-rw-r--r-- kimoonkim/staff 2025 2017-12-18 13:09 spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9/bin/load-spark-env.sh
-rwxr-xr-x kimoonkim/staff 2989 2017-12-18 13:09 spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9/bin/pyspark
-rw-r--r-- kimoonkim/staff 1170 2017-12-18 13:09 spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9/bin/pyspark.cmd
```

Reply:
Ah, forgot to mention that the unpacked top-level dir is the only top level file. That's the precondition for `mv * ../spark-distro` to work:
```
/Tmp/spark-distro-tmp$ tar xfz ../spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9.tgz
~/Tmp/spark-distro-tmp$ ls
spark-2.3.0-SNAPSHOT-bin-20171218-772e4648d9
```